.github/workflows/npm-sonar-analysis.yml

```diff
@@ -67,23 +70,55 @@ jobs:
         echo "BRANCH_NAME=$(echo ${{ github.ref }} | sed -e 's,.*/\(.*\),\1,')" >> $GITHUB_ENV
         echo "GPG_TTY=$(tty)" >> $GITHUB_ENV
 
-    - name: run sonar analysis
+    - name: setup sonar properties
       run: |
+        if [[ -z $SONAR_TOKEN ]]; then
+          echo "\$SONAR_TOKEN environmental variable not set; EXITING;";
+          exit 1;
+        fi
+        if [[ -z $ORG_KEY ]]; then
+          echo "\$ORG_KEY environmental variable not set; EXITING;";
+          exit 1;
+        fi
+        if [[ -z $SONAR_URL ]]; then
+          echo "\$SONAR_URL environmental variable not set; EXITING;";
+          exit 1;
+        fi
+  
         cd "./${{inputs.SERVICE_LOCATION}}"
-        echo "sonar.host.url=${{ inputs.SONAR_URL }}
-              sonar.login=${{ secrets.SONAR_TOKEN }}
-              sonar.projectKey=${{ inputs.PROJECT_KEY }}
-              sonar.organization=${{ secrets.ORG_KEY }}
-              sonar.sourceEncoding=UTF-8
-              sonar.sources=src
-              sonar.exclusions=**/node_modules/**
-              sonar.tests=src
-              sonar.test.inclusions=**/*.spec.ts
-              sonar.typescript.lcov.reportPaths=coverage/lcov.info" >> sonar-project.properties
+        
+        repo_name="${{ github.event.repository.name }}"
```

Comment:
Instead of hardcoding the env variables for Inji specifically, lets just check if these variables are assigned a value or not. 
If value is assigned use the same.
In inji repo while calling this action lets pass these env variables as an input.
This way we make sure the required caller workflow's pass the values if needed. 
In case the same is not needed, it will analyse without these values.

Reply:
@ckm007 Can we design it like
 ```
if({{ inputs.SONAR-PROPERTIES-ALREADY-EXISTS }})
       //add only host url , login, projectKey, etc (commons - same as if part)
else 
      //add host url , login, projectKey, etc (commons) + exclusions, tests, etc (specifics)
```
(By default SONAR-PROPERTIES-ALREADY-EXISTS  is false)
- This way the caller can write their customisations in sonar-project.properties, no need of sending the required values as inputs / env. 
- Also, Projects will be already having their sonar-project.properties already defined which would be used during their local sonar analysis, can we make use of them?  
- Full customization is allowed here
- We need not check if the values is passed as env or not

Reply:
Changes made to transform the sonar config to inputs